include/tester/core/Application.hpp

```diff
@@ -60,15 +54,21 @@ class Application
 
     bsoncxx::document::value getHistoryHeaders()
     {
-        return _database->getAllHeaders();
+        return _database->getTestLogsHistory();
     }
 
-    bsoncxx::document::value getHistoryLog(const std::string &docId)
+    bsoncxx::document::value getHistoryLog(const std::string &documentId)
```

Comment:
check if passing this parameter by value and using `std::move` works.

Reply:
I can't it doesn't work.
I got this error:
Building file: build/Main.o
In file included from include/tester/core/Application.hpp:4,
                 from src/Main.cpp:1:
/usr/include/c++/11/functional: In instantiation of ‘constexpr _Result std::_Bind<_Functor(_Bound_args ...)>::__call(std::tuple<_Args2 ...>&&, std::_Index_tuple<_Indexes ...>) [with _Result = void; _Args = {std::vector<unsigned char, std::allocator<unsigned char> >&&}; long unsigned int ..._Indexes = {0, 1}; _Functor = void (TCPSession::*)(std::vector<unsigned char>); _Bound_args = {TCPSession*, std::_Placeholder<1>}]’:
/usr/include/c++/11/functional:503:32:   required from ‘constexpr _Result std::_Bind<_Functor(_Bound_args ...)>::operator()(_Args&& ...) [with _Args = {std::vector<unsigned char, std::allocator<unsigned char> >}; _Result = void; _Functor = void (TCPSession::*)(std::vector<unsigned char>); _Bound_args = {TCPSession*, std::_Placeholder<1>}]’
/usr/include/c++/11/bits/invoke.h:61:36:   required from ‘constexpr _Res std::__invoke_impl(std::__invoke_other, _Fn&&, _Args&& ...) [with _Res = void; _Fn = std::_Bind<void (TCPSession::*(TCPSession*, std::_Placeholder<1>))(std::vector<unsigned char>)>&; _Args = {std::vector<unsigned char, std::allocator<unsigned char> >}]’
/usr/include/c++/11/bits/invoke.h:111:28:   required from ‘constexpr std::enable_if_t<is_invocable_r_v<_Res, _Callable, _Args ...>, _Res> std::__invoke_r(_Callable&&, _Args&& ...) [with _Res = void; _Callable = std::_Bind<void (TCPSession::*(TCPSession*, std::_Placeholder<1>))(std::vector<unsigned char>)>&; _Args = {std::vector<unsigned char, std::allocator<unsigned char> >}; std::enable_if_t<is_invocable_r_v<_Res, _Callable, _Args ...>, _Res> = void]’
/usr/include/c++/11/bits/std_function.h:290:30:   required from ‘static _Res std::_Function_handler<_Res(_ArgTypes ...), _Functor>::_M_invoke(const std::_Any_data&, _ArgTypes&& ...) [with _Res = void; _Functor = std::_Bind<void (TCPSession::*(TCPSession*, std::_Placeholder<1>))(std::vector<unsigned char>)>; _ArgTypes = {std::vector<unsigned char, std::allocator<unsigned char> >}]’
/usr/include/c++/11/bits/std_function.h:451:21:   required from ‘std::function<_Res(_ArgTypes ...)>::function(_Functor&&) [with _Functor = std::_Bind<void (TCPSession::*(TCPSession*, std::_Placeholder<1>))(std::vector<unsigned char>)>; _Constraints = void; _Res = void; _ArgTypes = {std::vector<unsigned char, std::allocator<unsigned char> >}]’
include/tester/core/../vtcp/tcp/TCPSession.hpp:27:9:   required from here
/usr/include/c++/11/functional:423:9: internal compiler error: Segmentation fault
  423 |         }
      |         ^
0x7f12a0e4251f ???
        ./signal/../sysdeps/unix/sysv/linux/x86_64/libc_sigaction.c:0
0x7f12a0e29d8f __libc_start_call_main
        ../sysdeps/nptl/libc_start_call_main.h:58
0x7f12a0e29e3f __libc_start_main_impl
        ../csu/libc-start.c:392
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <file:///usr/share/doc/gcc-11/README.Bugs> for instructions.
make: *** [Makefile:132: build/Main.o] Error 1

Reply:
Done